nixos/modules/services/monitoring/librenms.nix

```diff
@@ -0,0 +1,670 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.librenms;
+
+  boolToString = x: (if x then "true" else "false");
+
+  package = pkgs.librenms.override {
+    logDir = cfg.logDir;
+    dataDir = cfg.dataDir;
+  };
+
+  phpPackage = pkgs.php82.buildEnv {
+    extraConfig = ''
+      log_errors = on
+      post_max_size = 100M
+      upload_max_filesize = 100M
+      date.timezone = "${config.time.timeZone}"
+    '';
+  };
+
+  pythonPackages = ps:
+    with ps; [
+      pymysql
+      python-dotenv
+      redis
+      setuptools
+      psutil
+      command_runner
+    ];
+
+  artisanWrapper = pkgs.writeShellScriptBin "artisan" ''
+    cd ${package}
+    sudo=exec
+    if [[ "$USER" != ${cfg.user} ]]; then
+      sudo='exec /run/wrappers/bin/sudo -u ${cfg.user}'
+    fi
+    $sudo ${phpPackage}/bin/php artisan $*
+  '';
+
+  lnmsWrapper = pkgs.writeShellScriptBin "lnms" ''
+    cd ${package}
+    exec ${phpPackage}/bin/php lnms $*
+  '';
+
+  configFile = pkgs.writeText "config.php" (''
+    <?php
+    $config['user'] = "${cfg.user}";
+    $config['own_hostname'] = "${cfg.hostname}";
+    $config['base_url'] = "${cfg.config.baseUrl}";
+    $config['auth_mechanism'] = "${cfg.config.authMechanism}";
+
+    # disable auto update function (won't work with NixOS)
+    $config['update'] = false;
+
+    # enable fast ping by default
+    $config['ping_rrd_step'] = 60;
+
+    # one minute polling
+    $config['rrd.step'] = ${if cfg.enableOneMinutePolling then "60" else "300"};
+    $config['rrd.heartbeat'] = ${if cfg.enableOneMinutePolling then "120" else "600"};
+
+    $config['fping'] = "/run/wrappers/bin/fping";
+    $config['mtr'] = "${config.programs.mtr.package}/bin/mtr";
+    $config['rrdtool'] = "${pkgs.rrdtool}/bin/rrdtool";
+    $config['snmpwalk'] = "${pkgs.net-snmp}/bin/snmpwalk";
+    $config['snmpget'] = "${pkgs.net-snmp}/bin/snmpget";
+    $config['snmpgetnext'] = "${pkgs.net-snmp}/bin/snmpgetnext";
+    $config['snmpbulkwalk'] = "${pkgs.net-snmp}/bin/snmpbulkwalk";
+
+    ${optionalString (cfg.config.extraConfig != null) cfg.config.extraConfig}
+  '' + optionalString (cfg.config.authMechanism == "ldap") ''
```

Comment:
Maybe an LDAP settings object would be nicer, probably requires less maintenance and less code.

Reply:
Fixed with [RFC0042](https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md) refactoring.